### PR TITLE
AuthHelpers.correct_user?/2 should support email user validation

### PR DESCRIPTION
**Pull request by fteem**
Related to #358  and #413

`AuthHelpers.correct_user?/2` should be able to validate users based on username as well as email. This change makes authorization comply with changes done on the aforementioned issue/PR.

### CODE REVIEW

**Comment by fteem** [comment]
P.S. [CI build](https://travis-ci.org/hexpm/hex_web/builds/163832959) fails with a rather weird error:

```
* Getting db_connection (Hex package)
  Checking package (https://repo.hex.pm/tarballs/db_connection-1.0.0-rc.5.tar)
Request failed (:socket_closed_remotely)
** (Mix) Package fetch failed and no cached copy available
The command "mix deps.get" failed and exited with 1 during .
```

Any known remedy? 😃

File: web/controllers/auth_helpers.ex [hunk 1]
```diff
@@ -103,6 +103,6 @@ defmodule HexWeb.AuthHelpers do
 
   def correct_user?(%Plug.Conn{} = conn, user),
     do: correct_user?(conn.params["name"], user)
-  def correct_user?(name, user) when is_binary(name),
-    do: name == user.username
+  def correct_user?(username_or_email, user) when is_binary(username_or_email),
+    do: Enum.member?([user.username, user.email], username_or_email)
```
Comment (ericmj) [inline web/controllers/auth_helpers.ex]: `username_or_email in [user.username, user.email]`

**Comment by ericmj** [comment]
> Any known remedy?

Pray that someone fixes httpc :(

**Comment by ericmj** [comment]
Thank you @fteem! :heart:

File: web/controllers/auth_helpers.ex (hunk 1 above)
Comment (fteem) [inline web/controllers/auth_helpers.ex]: That is a good suggestion... my Ruby leaked in :)

**Pull request merged**